w3-provider.md

```diff
@@ -13,463 +12,353 @@
 
-## `consumer/add`
+```ipldsch
+type SubscriptionState union {
+  PendingSubscription     "pending"
+  ActiveSubscription      "active"
+} representation keyed
+```
 
-An agent MUST be delegated `consumer/add` capability, on successful [`provider/get`] invocation.
+#### Pending Subscription
 
-> Please note that provider MAY also delegate `consumer/add` capability for no reason at all e.g. as free giveaway campaign.
+[Provider] MUST return `PendingSubscription` variant when it requires an out-of-bound interaction to start a subscription. For example provider MAY require payment setup for the billing.
 
-```json
-{
-  "iss": "did:web:web3.storage",
-  "aud": "did:key:zAgent",
-  "att": [
-    {
-      "can": "consumer/add",
-      "with": "did:web:web3.storage:plan:free",
-      "nb": {
-        // link to a "provider/get" invocation
-        "request": { "/": "bafy...get" },
-        // did of the consumer space
-        "consumer": "did:key:zSpace"
-      }
-    }
-  ],
-  "prf": [
-    {
-      "iss": "did:web:web3.storage:plan:free",
-      "aud": "did:web:web3.storage",
-      "att": [
-        {
-          "can": "consumer/add",
-          "with": "did:web:web3.storage:plan:free"
-        }
-      ]
-    }
-  ]
+##### Pending Subscription Schema
+
+```ipldsch
+type PendingSubscription struct {
+  provider      ProviderDID
+  order         String
+  url           URLString
 }
-```
 
-### add `aud`
+type URLString = string
+type ProviderDID = DID
+```
 
-Capability MUST be delegated back to the `iss` of the [`provider/get`] request.
+##### Pending Subscription URL
 
-> This allows authorized agent to delegate `provider/get` capability to an agent or another user, which can then complete the loop using [`consumer/add`]. If capability was delegated back to `with` identifier instead, only account or delegate (with `consumer/add` capability) would be able to complete the loop.
+The `url` field MUST be set to the location user is expected to navigate, to complete subscription process.
 
-### add `with`
+##### Pending Subscription Provider
 
-Capability resource MUST be a provider DID. It MUST be the same as [`nb.provider`](#get-nbprovider) of the corresponding [`provider/get`] invocation.
+The `provider` field MUST be set to the [DID] of the [subscription] [provider].
 
-### add `nb.consumer`
+#### Active Subscription
 
-The `nb.consumer` MUST be set to the same DID as [`nb.consumer`](#get-nbconsumer) of the [`provider/get`] request.
+[Provider] MUST return `ActiveSubscription` variant when out-of-bound interaction is not required. For example provider may offer a free plan to any [account]. Provider could return `ActiveSubscription` if it already has billing info for the [account].
 
-> ⚠️ Omitting `nb.consumer` allows delegate to add arbitrary number of consumers to the provider
+##### Active Subscription Schema
 
-### add `nb.request`
+```ipldsch
+type ActiveSubscription struct {
+  provider          ProviderDID
+  product           ProductDID
```

Comment:
```suggestion
  product           URL
```

Reply:
How about we just make it a URL, not strictly necessary either but seems like a nudge into a right direction so implementers are forced to namespace